src/compiler/parser.ts

```diff
@@ -6472,36 +6490,66 @@ namespace ts {
                         parseExpected(SyntaxKind.CloseBracketToken);
                     }
 
-                    return { name, isBracketed };
+                    return { fullName, isBracketed };
                 }
 
-                function parseParameterOrPropertyTag(atToken: AtToken, tagName: Identifier, shouldParseParamTag: boolean): JSDocPropertyTag | JSDocParameterTag {
+                function isObjectOrObjectArrayTypeReference(node: TypeNode): boolean {
+                    return node.kind === SyntaxKind.ObjectKeyword ||

```

Comment:
`isTypeReferenceNode` just tests the syntax kind, so just make that a switch case and have `default: return false;`.